languages/python/oso/polar/ffi.py

```diff
@@ -88,43 +99,50 @@ def call_result(self, call_id, value):
             value = ffi.NULL
         else:
             value = ffi_serialize(value)
-        check_result(lib.polar_call_result(self.ptr, call_id, value))
+        self.check_result(lib.polar_call_result(self.ptr, call_id, value))
 
     def question_result(self, call_id, answer):
         answer = 1 if answer else 0
-        check_result(lib.polar_question_result(self.ptr, call_id, answer))
+        self.check_result(lib.polar_question_result(self.ptr, call_id, answer))
 
     def application_error(self, message):
         """Pass an error back to polar to get stack trace and other info."""
         message = to_c_str(message)
-        check_result(lib.polar_application_error(self.ptr, message))
+        self.check_result(lib.polar_application_error(self.ptr, message))
 
     def next_event(self):
         event = lib.polar_next_query_event(self.ptr)
-        process_messages(self.next_message)
-        event = check_result(event)
+        self.process_messages()
+        event = self.check_result(event)
         return QueryEvent(event)
 
     def debug_command(self, command):
         result = lib.polar_debug_command(self.ptr, ffi_serialize(command))
-        process_messages(self.next_message)
-        check_result(result)
+        self.process_messages()
+        self.check_result(result)
 
     def next_message(self):
         return lib.polar_next_query_message(self.ptr)
 
     def source(self):
         source = lib.polar_query_source_info(self.ptr)
-        source = check_result(source)
+        source = self.check_result(source)
         return Source(source)
 
     def bind(self, name, value):
         name = to_c_str(name)
         value = ffi_serialize(value)
         result = lib.polar_bind(self.ptr, name, value)
         # TODO(gj): Do we need to process_messages here?
-        process_messages(self.next_message)
-        check_result(result)
+        self.process_messages()
+        self.check_result(result)
+
+    def check_result(self, result):
+        return check_result(result, lambda msg: self.host.enrich_message(msg))
+
+    def process_messages(self):
```

Comment:
Let's add an assert here that `self.host is not None` to catch forgetting to set the `self.host` property after the constructor. This will help us catch that even if we don't write a test that emits messages.